# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RemoteImageHolder.cpp
# Commit: afef5857aea6
# Full Hash: afef5857aea6cbbd3e9296ce39c004812967d095
# Author: Brad Werth <bwerth@mozilla.com>
# Date: 2022-09-08 04:52:24
# Regressor Bug: 1787561
# File Overlap Count: 1
# Description:
#   Bug 1787561 Part 1: Make GPUVideoImage track color depth, and make RemoteImageHolder pass it through. r=media-playback-reviewers,gfx-reviewers,alwu,aosmond
#   
#   This ensures that the Image that the client receives has the color depth
#   information that was sent to the GPU, even though the client can't query
#   the GPU directly.
# ==============================================================================

diff -r 1b2dc91d3c38 -r afef5857aea6 dom/media/ipc/RemoteImageHolder.cpp
--- a/dom/media/ipc/RemoteImageHolder.cpp	Wed Sep 07 18:23:24 2022 +0000
+++ b/dom/media/ipc/RemoteImageHolder.cpp	Wed Sep 07 18:31:45 2022 +0000
@@ -22,11 +22,17 @@
 RemoteImageHolder::RemoteImageHolder(layers::IGPUVideoSurfaceManager* aManager,
                                      layers::VideoBridgeSource aSource,
                                      const gfx::IntSize& aSize,
+                                     const gfx::ColorDepth& aColorDepth,
                                      const layers::SurfaceDescriptor& aSD)
-    : mSource(aSource), mSize(aSize), mSD(Some(aSD)), mManager(aManager) {}
+    : mSource(aSource),
+      mSize(aSize),
+      mColorDepth(aColorDepth),
+      mSD(Some(aSD)),
+      mManager(aManager) {}
 RemoteImageHolder::RemoteImageHolder(RemoteImageHolder&& aOther)
     : mSource(aOther.mSource),
       mSize(aOther.mSize),
+      mColorDepth(aOther.mColorDepth),
       mSD(std::move(aOther.mSD)),
       mManager(aOther.mManager) {
   aOther.mSD = Nothing();
@@ -113,7 +119,7 @@
     SurfaceDescriptorRemoteDecoder remoteSD =
         static_cast<const SurfaceDescriptorGPUVideo&>(*mSD);
     remoteSD.source() = Some(mSource);
-    image = new GPUVideoImage(mManager, remoteSD, mSize);
+    image = new GPUVideoImage(mManager, remoteSD, mSize, mColorDepth);
   }
   mSD = Nothing();
   mManager = nullptr;
@@ -139,6 +145,7 @@
     RemoteImageHolder&& aParam) {
   WriteIPDLParam(aWriter, aActor, aParam.mSource);
   WriteIPDLParam(aWriter, aActor, aParam.mSize);
+  WriteIPDLParam(aWriter, aActor, aParam.mColorDepth);
   WriteIPDLParam(aWriter, aActor, aParam.mSD);
   // Empty this holder.
   aParam.mSD = Nothing();
@@ -150,6 +157,7 @@
     RemoteImageHolder* aResult) {
   if (!ReadIPDLParam(aReader, aActor, &aResult->mSource) ||
       !ReadIPDLParam(aReader, aActor, &aResult->mSize) ||
+      !ReadIPDLParam(aReader, aActor, &aResult->mColorDepth) ||
       !ReadIPDLParam(aReader, aActor, &aResult->mSD)) {
     return false;
   }